ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/metrics/ConsumptionSpeedMetricSpec.scala

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.benchtool
+
+import com.daml.ledger.api.benchtool.metrics.Metric.ConsumptionSpeedMetric
+import com.google.protobuf.timestamp.Timestamp
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+import java.time.{Clock, Instant}
+import scala.util.Random
+
+class ConsumptionSpeedMetricSpec extends AnyWordSpec with Matchers {
+  ConsumptionSpeedMetric.getClass.getSimpleName should {
+    "correctly handle initial state" in {
+      val metric = ConsumptionSpeedMetric.empty[String](100, dummyRecordTimesFunction)
+
+      val (_, periodicValue) = metric.periodicValue()
+      val finalValue = metric.finalValue(1.0)
```

Comment:
This is a pure function disguised as a class member method. You can only use it if you have an instance of `ConsumptionSpeedMetric` which is kind of strange, since you don't actually need an instance of anything since it doesn't depend on the context.

Reply:
The `def finalValue()` method is a pure function in some metrics, but there are others that implement the function using their internal state. That's why it's a part of `Metric` trait.

Reply:
I see, fine then 👍